tutorials/multimodal/NeVA Tutorial.ipynb

```diff
@@ -0,0 +1,249 @@
+    " model.data.image_folder=/path/to/dataset/LLaVA-Pretrain-LCS-558K/images \\\n",
+    " model.tokenizer.library=megatron \\\n",
+    " model.tokenizer.model=/path/to/tokenizer \\\n",
+    " model.encoder_seq_length=4096 \\\n",
+    " model.num_layers=24 \\\n",
+    " model.hidden_size=2048 \\\n",
+    " model.ffn_hidden_size=5440 \\\n",
+    " model.num_attention_heads=16 \\\n",
+    " model.normalization=layernorm1p \\\n",
+    " model.do_layer_norm_weight_decay=False \\\n",
+    " model.apply_query_key_layer_scaling=True \\\n",
+    " model.activation=fast-swiglu \\\n",
+    " model.headscale=False \\\n",
+    " model.position_embedding_type=rope \\\n",
+    " model.rotary_percentage=0.5 \\\n",
+    " model.num_query_groups=null \\\n",
+    " model.data.num_workers=8 \\\n",
+    " model.mm_cfg.llm.from_pretrained=/path/to/checkpoint \\\n",
+    " model.mm_cfg.llm.model_type=nvgpt \\\n",
+    " exp_manager.create_checkpoint_callback=True \\\n",
+    " model.data.conv_template=${model.mm_cfg.llm.model_type} \\\n",
+    " model.mm_cfg.vision_encoder.from_pretrained='openai/clip-vit-large-patch14' \\\n",
+    " model.mm_cfg.vision_encoder.from_hf=True \\\n",
+    " model.data.image_token_len=256 \\\n",
+    " model.optim.name=\"fused_adam\" \\\n",
+    "```\n",
+    "\n",
+    "## Inference\n",
+    "\n",
+    "### From Pre-trained Checkpoints\n",
+    "\n",
+    "If you would like to use NeVA for inference from pre-trained checkpoint via HuggingFace, you can convert from HuggingFace to `.nemo` first.\n",
+    "\n",
+    "First, download the model checkpoint from HuggingFace [here](https://huggingface.co/liuhaotian/llava-v1.5-7b). The tokenizer (stored as `tokenizer.model` within the pretrained checkpoint) must be modified with the following commands:\n",
+    "\n",
+    "```\n",
+    "cd /opt/sentencepiece/src/\n",
+    "protoc --python_out=/opt/NeMo/scripts/tokenizers/ sentencepiece_model.proto\n",
+    "python /opt/NeMo/scripts/tokenizers/add_special_tokens_to_sentencepiece.py \\\n",
+    "--input_file /path/to/tokenizer.model \\\n",
+    "--output_file /path/to/tokenizer_neva.model \\\n",
+    "--is_userdefined \\\n",
+    "--tokens \"<extra_id_0>\" \"<extra_id_1>\" \"<extra_id_2>\" \"<extra_id_3>\" \\\n",
+    "         \"<extra_id_4>\" \"<extra_id_5>\" \"<extra_id_6>\" \"<extra_id_7>\"\n",
+    "```\n",
+    "\n",
+    "Finally, convert to `.nemo` via the provided script:\n",
+    "\n",
+    "```\n",
+    "python /opt/NeMo/examples/multimodal/mllm/neva/convert_hf_llava_to_neva.py \\\n",
+    "--in-file /path/to/llava-v1.5-7b \\\n",
+    "--out-file /path/to/llava-v1.5-7b.nemo \\\n",
+    "--tokenizer-model /path/to/tokenizer_neva.model\n",
+    "```\n",
+    "\n",
+    "### Running Inference\n",
+    "\n",
+    "NeVA inference via the NeMo framework can be quickly spun up via the NeMo Launcher and a few modifications to use the default NeVA inference config file.\n",
+    "\n",
+    "#### Inference Config Setup\n",
```

Comment:
With the tutorial, we should provide a command to run neva inference directly.
I wonder if you got time to update this part?
This is an example,  

torchrun --nproc_per_node=1 /opt/NeMo/examples/multimodal/multimodal_llm/neva/neva_evaluation.py \
   tensor_model_parallel_size=1 \
   pipeline_model_parallel_size=1 \
   neva_model_file=/lustre/fsw/coreai_dlalgo_genai/datasets/checkpoints/llava-v1.5/llava-v1.5-13b.nemo \
   trainer.devices=1 \
   prompt_file=/lustre/fsw/coreai_dlalgo_genai/datasets/llava-bench-in-the-wild/input_llama_format.jsonl \
   inference.images_base_path=/lustre/fsw/coreai_dlalgo_genai/datasets/llava-bench-in-the-wild/images \
   output_file=/lustre/fsw/coreai_dlalgo_genai/datasets/llava-bench-in-the-wild/results/llava_v1_5_13b_converted.jsonl \
  inference.temperature=0.2 \
  inference.greedy=False \
  inference.add_BOS=False \
  inference.tokens_to_generate=10

Reply:
Added!